apps/backend/src/routes/signup/+page.svelte

```diff
@@ -0,0 +1,243 @@
+  import { createForm } from 'svelte-forms-lib';
+
+  import { authStore } from '$lib/stores';
+  import type { FirebaseError } from 'firebase/app';
+
+  // import sign up function
+  import { signOut, signUpEmailPassword } from '$lib/firebase/auth';
+
+  //import yup for form validation
+  import * as yup from 'yup';
+
+  // JavaScript code for Continue button
+  let apiError: FirebaseError | null;
+
+  // JavaScript code for Checkbox button
+  let trigger = false;
+
+  // Create a reference for the password
+  const passwordRef = yup.ref('password');
+
+  function toggleCheckbox() {
+    trigger = !trigger;
+  }
+
+  // Yup validation code
+  const { form, errors, handleChange, handleSubmit } = createForm({
+    initialValues: {
+      email: '',
+      password: '',
+      confirmPassword: '',
+    },
+    validationSchema: yup.object().shape({
+      email: yup.string().email().required('Corporate Email is required.'),
+      password: yup
+        .string()
+        .required('Password is required.')
+        .matches(
+          /^(?=.*[A-Za-z])(?=.*\d)(?=.*[@$!%*?&])[A-Za-z\d@$!%*?&]{8,}$/,
+          'Password must be at least 8 characters long and include at least one letter, one number, and one special character.'
+        ),
+      confirmPassword: yup
+        .string()
+        .required('Confirm Password is required.')
+        .oneOf([passwordRef], 'Passwords entered do not match.')
+        .matches(
+          /^(?=.*[A-Za-z])(?=.*\d)(?=.*[@$!%*?&])[A-Za-z\d@$!%*?&]{8,}$/,
+          'Password must be at least 8 characters long and include at least one letter, one number, and one special character.'
+        ),
+    }),
+    onSubmit: async (data) => {
+      if (data.password === data.confirmPassword) {
+        // This will only be executed if passwords match
+        await signUpEmailPassword(data.email, data.password, (error) => (apiError = error));
+        console.log('Account created successfully');
+      } else {
+        console.error('Passwords do not match');
+      }
+    },
+  });
+</script>
+
+<svelte:head>
+  <title>Sign Up</title>
+</svelte:head>
+
+<!-- Armadillo Logo-->
+<div class="flex items-center justify-center p-10">
+  <img src="armadillo.png" alt="" />
+</div>
+
+<div class="flex flex-row items-center justify-center">
+  <!-- Signup Form - Card -->
+  <div class="w-full max-w-2xl p-10 border border-secondary rounded-md shadow-lg">
+    <!-- Signup Form - Header -->
+    <h1 class="text-3xl font-semibold text-center mb-5">Create your account</h1>
+
+    <form on:submit|preventDefault={handleSubmit}>
+      <!-- General Error Message -->
+      {#if apiError && apiError.code === 'auth/email-already-in-use'}
+        <div class="p-4 flex flex-row bg-red-50 text-red-800 rounded-lg" role="alert">
+          <img src="errorlogo.svg" alt="" />
+          <div class="ml-2 text-sm font-medium"
+            >You're too fast, slow down and try again in a while</div>
+        </div>
+      {/if}
```

Comment:
`auth/email-already-in-use` does not mean that they're being rate-limited

Reply:
Noted, changed to a more generic message -> `Invalid Email or Password, Please try again.`
Only copied the message from dr. receipts but left it untouched, my bad.